code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/buildscript/OpendRootBuildScript.xtend

```diff
@@ -0,0 +1,96 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.buildscript
+
+import de.fhdo.lemma.model_processing.code_generation.container_base.template.BuildScriptTemplate
+import java.util.Map
+import java.util.HashMap
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.util.regex.Pattern
+
+/**
+ * This class contains the logic for the generation of the build scripts, used in the
+ * [BuildScriptCodeGenerator].
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+class OpendRootBuildScript {
+    static OpendRootBuildScript instance
+    Map<String, String> buildScriptNodes
+
+    new () {
+        this.buildScriptNodes = new HashMap
+    }
+
+    /*
+     * Get or create instance of the [OpendRootBuildScript] class.
+     */
+    static def OpendRootBuildScript getInstance() {
+        if (OpendRootBuildScript.instance === null)
+            OpendRootBuildScript.instance = new OpendRootBuildScript
+
+        return OpendRootBuildScript.instance
+    }
+
+    /*
+     * Load a  existing root build script from the file system into the data structure of the
+     * [OpendRootBuildScript] class.
+     * Note: The name of the nodes is extracted from the content of each line of the script via
+     * a regex. This may lead to a problem with build scripts not crated by this code generator.
+     */
+    def loadExistingRootBuildScript(String targetFolder) {
+        val file = new File(targetFolder + File.separator + "build" + File.separator + "build.sh")
+        if (file.exists) {
+            var stream = Files.lines(
+                Paths.get(targetFolder
+                    + File.separator
+                    + "build"
+                    + File.separator
+                    + "build.sh"),
+                StandardCharsets.UTF_8
+            )
+            stream.forEach[line |
+                this.addOrReplace(this.getNodeNameFromBuildScriptLine(line), line)
+            ]
+            buildScriptNodes.remove("usrbinenv")
+        }
+    }
+
+    /*
+     * Add or replace the build command based the name of the operation node.
+     */
+    def addOrReplace(String nodeName, String buildCommand) {
+        buildScriptNodes.remove(nodeName.toLowerCase)
```

Comment:
Why is the removal necessary? Just always do a `put` to add or update a non-existing/existing entry (https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#put-K-V-).